core/src/main/java/org/opensearch/sql/analysis/model/CatalogName.java

```diff
@@ -14,7 +14,7 @@
 @Getter
 public class CatalogName {
 
-  public static final String DEFAULT_CATALOG_NAME = ".opensearch";
+  public static final String DEFAULT_CATALOG_NAME = "@opensearch";
   private String name = DEFAULT_CATALOG_NAME;
```

Comment:
Why this is a field member? Are we passing state between `capture` function call?

Reply:
Yes. Kind of

`prometheus.information_schema.tablename`

CatalogName captures prometheus into `name` variable and returns the remaining list `information_schema.tablename`

SchemaName captures information_schema and returns the remaining list tablename.

Remaining string is treated as table name.

Reply:
Will update class description. Let me know if you have any other thoughts.

Reply:
I see. I was confused. I think this `capture()` work should be done in constructor instead?

Reply:
Right...if some creates the class and calls getName it would give wrong details. Let me correct.

Reply:
Refactored to a single CatalogSchemaIdentifierNameResovler class



Removed CatalogName and SchemaNameClasses and introduced new methods captureSchemaName and captureCatalogName in the resolver class.